Deployetl

szinck1 opened the pull request:
Deploy the ETL to Docker Hub, tag images with SHA1 instead of $VERSION.

szinck1 committed:
deploy ETL, remove VERSION and tag imges with SHA1

szinck1 committed:
set working dir to etl for deploy_etl

JasonMWhite commented on .circleci/config.yml:
```diff
@@ -1,92 +1,87 @@
-        run:
+      - run: 
           command: |
               mkdir ~/repo/api/dist
```
This whole section might be more palatable if we put it in a bash file in `etl/scripts`?

JasonMWhite commented on .circleci/config.yml:
```diff
@@ -1,92 +1,87 @@
-    docker:
-      -
+jobs: 
+  deploy_api: 
```
Do we need 3 separate deployment jobs? Could we do it with 1 that deploys all three sections? Would use less containers on CI, of which we only have 4.

JasonMWhite commented on .circleci/config.yml:
```diff
@@ -1,92 +1,87 @@
-              docker build --build-arg VERSION="${VERSION}" -t "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"nrcan_endpoint":"${VERSION}" -t "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"nrcan_endpoint":"latest" .
-              docker push "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"nrcan_endpoint":"${VERSION}"
-              docker push "${DOCKER_REGISTRY}"/"${NAMESPACE}"/"nrcan_endpoint":"latest"
+              docker build -t "${DOCKER_REGISTRY}/${NAMESPACE}/nrcan_etl:${CIRCLE_SHA1}" -t "${DOCKER_REGISTRY}/${NAMESPACE}/nrcan_etl:latest" .
```
These are all so similar, it would seem they could all be a single bash file with different arguments passed in.

JasonMWhite commented on .circleci/config.yml:
```diff
@@ -1,92 +1,87 @@
+      - save_cache: 
           key: "deps1-{{ .Branch }}-{{ checksum \"requirements.txt\" }}"
-          paths:
+          paths: 
```
extra whitespace

JasonMWhite commented on .circleci/config.yml:
```diff
@@ -100,255 +95,215 @@ jobs:
+      - save_cache: 
           key: "deps2-{{ .Branch }}-{{ checksum \"requirements.txt\" }}"
-          paths:
+          paths: 
```
whitespace

JasonMWhite commented on .circleci/config.yml:
```diff
@@ -100,255 +95,215 @@ jobs:
-      -
-        checkout:
+    steps: 
+      - checkout: 
```
I appreciate the reformatting, but generally it would be better to do that in a separate, prior PR. Makes it easier to identify the critical new parts from the lower-risk reformatting.

JasonMWhite commented on .circleci/config.yml:
```diff
@@ -100,255 +95,215 @@ jobs:
-  version: 2
-  node_and_python:
-    jobs:
+version: 2
```
Why move this down?

JasonMWhite commented on .circleci/config.yml:
```diff
@@ -100,255 +95,215 @@ jobs:
+version: 2
+workflows: 
+  node_and_python: 
+    jobs: 
```
whitespace

szinck1 committed:
remove unnecessary yarn_build job and clean up deploy_api

szinck1 committed:
removed root workspace by mistake

szinck1 committed:
removed root workspace by mistake

szinck1 commented:
I think I've addressed your concerns @JasonMWhite except for moving the deployment jobs into one big job. We probably could but I thought having them set up individually might give us more freedom to adjust parameters later on.  In any case, I wanted to address the other issues right away.

JasonMWhite commented:
Looks like the ETL tests aren't able to connect to MongoDB - connection refused.

szinck1 committed:
Merge branch 'master' of github.com:cds-snc/nrcan_api into deployetl

szinck1 committed:
Merge branch 'master' of github.com:cds-snc/nrcan_api into deployetl

szinck1 committed:
trigger build

buckley-w-david commented:
CI hasn't run on this PR for some reason

szinck1 committed:
remove :

buckley-w-david reviewed (changes requested):
Not approved until CI has run and succeeded

szinck1 committed:
add azurite image to etl job

szinck1 committed:
fix typo
